core/src/OpenMPTarget/Kokkos_OpenMPTarget_Instance.cpp

```diff
@@ -66,7 +66,29 @@ void OpenMPTargetInternal::fence(const std::string& name,
   }
 }
 int OpenMPTargetInternal::concurrency() const {
-  return 128000;  // FIXME_OPENMPTARGET
+  int max_threads = 2048 * 80;
+#if defined(KOKKOS_IMPL_ARCH_NVIDIA_GPU)
+  int max_threads_block = 2048;
+#if defined(KOKKOS_ARCH_AMPERE86)
+  max_threads = max_threads_block * 1536;
+#elif defined(KOKKOS_ARCH_AMPERE80)
+  max_threads = max_threads_block * 108;
+#elif defined(KOKKOS_ARCH_VOLTA72)
+  max_threads = max_threads_block * 84;
+#elif defined(KOKKOS_ARCH_VOLTA70)
+  max_threads = max_threads_block * 80;
+#elif defined(KOKKOS_ARCH_PASCAL60) || defined(KOKKOS_ARCH_PASCAL61)
+  max_threads = max_threads_block * 60;
+#endif
+#elif defined(KOKKOS_ARCH_INTEL_GPU)
+#pragma omp target map(max_threads)
+  { max_threads = omp_get_num_procs(); }
```

Comment:
Does this give you the number of EUs? If so, shouldn't the value be a little higher since multiple workgroups could be scheduled by a EU? See https://github.com/intel/llvm/blob/756ba2616111235bba073e481b7f1c8004b34ee6/sycl/source/detail/reduction.cpp#L51-L62.

Reply:
As per OpenMP spec, this should give us the number of procs on the device and each proc can then execute a SIMD instruction.